6001 : [기초-출력] 출력하기01(설명)(py)

python 언어에서 가장 기본적인 명령이 출력문이다.
print( )를 이용해 다음 단어를 출력하시오.

Hello

예시
print("Hello") 

In [1]:
print("Hello")

Hello


6002 : [기초-출력] 출력하기02(설명)(py)

이번에는 공백( )을 포함한 문장을 출력한다.
다음 문장을 출력해보자.

Hello World
(대소문자에 주의한다.)

In [2]:
print("Hello World")

Hello World


6003 : [기초-출력] 출력하기03(설명)(py)

이번에는 줄을 바꿔 출력하는 출력문을 연습해보자.
다음과 같이 줄을 바꿔 출력해야 한다.

Hello
World
(두 줄에 걸쳐 줄을 바꿔 출력)

예시 
print("Hello") 
print("World")

In [3]:
print("Hello") 
print("World")

Hello
World


6004 : [기초-출력] 출력하기04(설명)(py)

이번에는 작은 따옴표(')(single quotation mark)가 들어있는
출력문 연습을 해보자.

다음 문장을 출력하시오.

'Hello'

예시
print(" ' ");

In [6]:
print("'Hello'")

'Hello'


6005 : [기초-출력] 출력하기05(설명)(py)

이번에는 큰따옴표(")(double quotation mark)가 포함된 출력문을 연습해보자.

다음 문장을 출력하시오.

"Hello World"
(단, 큰따옴표도 함께 출력한다.)

예시
print(' " ');


In [7]:
print('"Hello World"')

"Hello World"


6006 : [기초-출력] 출력하기06(py)

이번에는 특수문자 출력에 도전하자!!

다음 문장을 출력하시오.

"!@#$%^&*()'
(단, 큰따옴표와 작은따옴표도 함께 출력한다.)

In [16]:
print("\"!@#$%^&*()'")

"!@#$%^&*()'


6007 : [기초-출력] 출력하기07(py)

윈도우 운영체제의 파일 경로를 출력하는 연습을 해보자.
 
파일 경로에는 특수문자들이 포함된다.

다음 경로를 출력하시오.

"C:\Download\'hello'.py"
(단, 따옴표도 함께 출력한다.)

In [44]:
print("\"C:\\Download\\'hello'.py\"")

"C:\Download\'hello'.py"


6008 : [기초-출력] 출력하기08(py)

출력문 연습의 마지막 문제이다.
(생각과 시도를 많이 해야하는 문제들은 한 두 문제씩 넘겼다가 나중에 풀어보면 된다.)

이번에는 다음과 같은 python프로그램의 소스코드를 출력해보자.

print("Hello\nWorld")

위 코드를 정확히 그대로 출력하시오.(공백문자 주의)

In [50]:
print('print("Hello\\nWorld")')

print("Hello\nWorld")


6009 : [기초-입출력] 문자 1개 입력받아 그대로 출력하기(설명)(py)

문자(character)는
0~9, a~z, A~Z, !, @, #, {, [, <, ... 과 같이 
길이가 1인 기호라고 할 수 있다.

변수에 문자 1개를 저장한 후
변수에 저장되어 있는 문자를 그대로 출력해보자.

예시
c = input()
print(c)

와 같은 형태로 가능하다.

In [53]:
inputs = input()
print(inputs)

a


6010 : [기초-입출력] 정수 1개 입력받아 int로 변환하여 출력하기(설명)(py)

정수(integer)는
양의 정수(1, 2, 3, 4, 5, ...), 음의 정수(-1, -2, -3, -4, -5, ...), 0 과 같이
소숫점 아래에 수가 없는 수라고 할 수 있다.

변수에 정수값을 저장한 후 정수로 변환하여 출력해보자.

예시
n = input()
n = int(n)
print(n)
와 같은 형태로 가능하다.

In [54]:
inputs = input()
print(int(inputs))

15
15


6011 : [기초-입출력] 실수 1개 입력받아 변환하여 출력하기(설명)(py)

숫자(0~9)와 소수점(.)을 사용해 표현한 수를 실수(real number)라고 한다.

변수에 실수값을 저장한 후
변수에 저장되어 있는 값을 그대로 출력해보자.

예시
f = input()
f = float(f)
print(f)
와 같은 형태로 가능하다.

In [55]:
inputs = input()
print(float(inputs))

1.414213
1.414213


6012 : [기초-입출력] 정수 2개 입력받아 그대로 출력하기1(설명)(py)

줄을 바꿔 정수(integer) 2개를 입력받아 줄을 바꿔 출력해보자.

예시
a = input() 
b = input()
print(a)
print(b)
과 같은 방법으로 가능하다.

In [56]:
inputs_01 = input()
inputs_02 = input()
print(inputs_01)
print(inputs_02)

1
2
1
2


6013 : [기초-입출력] 문자 2개 입력받아 순서 바꿔 출력하기1(py)

줄을 바꿔 문자(character) 2개를 입력받고, 순서를 바꿔 한 줄씩 출력해보자.

In [57]:
inputs_01 = input()
inputs_02 = input()
print(inputs_02)
print(inputs_01)

a
b
b
a


6014 : [기초-입출력] 실수 1개 입력받아 3번 출력하기(py)

실수(real number) 1개를 입력받아 줄을 바꿔 3번 출력해보자.

예시
...
print(f)  #f에 저장되어있는 값을 출력하고 줄을 바꾼다.
print(f)
print(f)
와 같은 방법으로 3번 줄을 바꿔 출력할 수 있다.

In [58]:
inputs = input()
print(float(inputs))
print(float(inputs))
print(float(inputs))

0.1
0.1
0.1
0.1


6015 : [기초-입출력] 정수 2개 입력받아 그대로 출력하기2(설명)(py)

공백을 두고 입력된정수(integer) 2개를 입력받아 줄을 바꿔 출력해보자.

예시
a, b = input().split()
print(a)
print(b)
과 같은 방법으로 두 정수를 입력받아 출력할 수 있다.

In [62]:
inputs_01, inputs_02 = input().split()
print(inputs_01)
print(inputs_02)

1 2
1
2


6016 : [기초-입출력] 문자 2개 입력받아 순서 바꿔 출력하기2(설명)(py)

공백을 두고 문자(character) 2개를 입력받아 순서를 바꿔 출력해보자.

In [63]:
inputs_01, inputs_02 = input().split()
print(inputs_02)
print(inputs_01)

1 2
2
1


6017 : [기초-입출력] 문장 1개 입력받아 3번 출력하기(설명)(py)

정수(integer), 실수, 문자(character), 문자열(string) 등 1개만 입력받아 한 줄로 3번 출력해보자.

예시
s = input()
print(s, s, s)  #공백으로 구분해 한 줄로 출력한다.
와 같은 방법으로 3번 출력할 수 있다.


In [65]:
inputs = input()
print(inputs, inputs, inputs) 

computer science
computer science computer science computer science


6018 : [기초-입출력] 시간 입력받아 그대로 출력하기(설명)(py)

24시간 시:분 형식으로 시간이 입력될 때, 그대로 출력하는 연습을 해보자.

예시
a, b = input().split(':')
print(a, b, sep=':')
와 같은 방법으로 가능하다.


In [68]:
inputs_01, inputs_02 = input().split(':')
print(inputs_01, inputs_02, sep=':')

3:16
3:16


6019 : [기초-입출력] 연월일 입력받아 순서 바꿔 출력하기(py)

"연도.월.일"을 입력받아 "일-월-연도" 순서로 바꿔 출력해보자.

In [69]:
inputs_01, inputs_02, inputs_03 = input().split('.')
print(inputs_03, inputs_02, inputs_01, sep='-')

2020.3.4
4-3-2020


6020 : [기초-입출력] 주민번호 입력받아 형태 바꿔 출력하기(py)
시간 제한: 1 Sec  메모리 제한: 128 MB

주민번호는 다음과 같이 구성된다.
XXXXXX-XXXXXXX

왼쪽 6자리는 생년월일(YYMMDD)이고, 오른쪽 7자리는 성별,출생지역,확인코드로 구성되어있다.
주민번호를 입력받아 형태를 바꿔 출력해보자.

In [70]:
inputs_01, inputs_02 = input().split('-')
print(inputs_01, inputs_02, sep='')

000907-1121112
0009071121112


6021 : [기초-입출력] 단어 1개 입력받아 나누어 출력하기(설명)(py)

알파벳과 숫자로 이루어진 단어 1개가 입력된다.
입력받은 단어의 각 문자를 한 줄에 한 문자씩 분리해 출력한다.

예시
s = input()
print(s[0])
print(s[1])

In [71]:
inputs = input()
print(inputs[0])
print(inputs[1])
print(inputs[2])
print(inputs[3])
print(inputs[4])

hello
h
e
l
l
o


6022 : [기초-입출력] 연월일 입력받아 나누어 출력하기(설명)(py)

6자리의 연월일(YYMMDD)을 입력받아 나누어 출력해보자.

참고
s = input()
print(s[0:2])


In [77]:
inputs = input()
print(inputs[:2], inputs[2:4], inputs[4:], sep=' ')

200304
20 03 04


6023 : [기초-입출력] 시분초 입력받아 분만 출력하기(py)

시:분:초 형식으로 시간이 입력될 때 분만 출력해보자.

어떻게 분만 출력해야 할지 주의 깊게 생각해야한다.


In [78]:
inputs_01, inputs_02, inputs_03 = input().split(':')
print(inputs_02)

17:23:57
23


6024 : [기초-입출력] 단어 2개 입력받아 이어 붙이기(설명)(py)

알파벳 문자와 숫자로 이루어진 단어 2개를 입력받아
순서대로 붙여 출력하는 프로그램을 작성해보자.

예시
w1, w2 = input().split()
s = w1 + w2
print(s)

In [79]:
inputs_01, inputs_02 = input().split(' ')
print(inputs_01, inputs_02, sep='')

hello world
helloworld


6025 : [기초-값변환] 정수 2개 입력받아 합 계산하기(설명)(py)

정수 2개를 입력받아
합을 출력하는 프로그램을 작성해보자.

예시
a, b = input().split()
c = int(a) + int(b)
print(c)

In [80]:
inputs_01, inputs_02 = input().split(' ')
calculate = int(inputs_01) + int(inputs_02)
print(calculate)

123 -123
0


6026 : [기초-값변환] 실수 2개 입력받아 합 계산하기(설명)(py)

실수 2개를 입력받아
합을 출력하는 프로그램을 작성해보자.

In [81]:
inputs_01 = input()
inputs_02 = input()
calculate = float(inputs_01) + float(inputs_02)
print(calculate)

0.1
0.9
1.0


6027 : [기초-출력변환] 10진 정수 입력받아 16진수로 출력하기1(설명)(py)

10진수를 입력받아 16진수(hexadecimal)로 출력해보자.

예시
a = input()
n = int(a)            #입력된 a를 10진수 값으로 변환해 변수 n에 저장
print('%x'% n)  #n에 저장되어있는 값을 16진수(hexadecimal) 소문자 형태 문자열로 출력

In [82]:
inputs = input()
print('%x'%int(inputs))

255
ff


6028 : [기초-출력변환] 10진 정수 입력받아 16진수로 출력하기2(설명)(py)

10진수를 입력받아 16진수(hexadecimal)로 출력해보자.

예시
print('%X' % n)  #n에 저장되어있는 값을 16진수 대문자 형태 문자열로 출력


In [83]:
inputs = input()
print('%X'%int(inputs))

255
FF


6029 : [기초-값변환] 16진 정수 입력받아 8진수로 출력하기(설명)(py)

16진수를 입력받아 8진수(octal)로 출력해보자.

예시
a = input()
n = int(a, 16)      #입력된 a를 16진수로 인식해 변수 n에 저장
print('%o' % n)  #n에 저장되어있는 값을 8진수(octal) 형태 문자열로 출력

In [84]:
inputs = input()
print('%o'%int(inputs, 16))

f
17


6030 : [기초-값변환] 영문자 1개 입력받아 10진수로 변환하기(설명)(py)

영문자 1개를 입력받아 10진수 유니코드(Unicode) 값으로 출력해보자.

예시
n = ord(input())
print(n)

In [86]:
inputs = input()
print(ord(inputs))

A
65


6031 : [기초-값변환] 정수 입력받아 유니코드 문자로 변환하기(설명)(py)

10진 정수 1개를 입력받아
유니코드 문자로 출력해보자.


예시
c = int(input())
print(chr(c))  #c에 저장되어 있는 정수 값을 유니코드 문자(chracter)로 바꿔 출력한다. 

In [87]:
inputs = int(input())
print(chr(inputs))

65
A


6032 : [기초-산술연산] 정수 1개 입력받아 부호 바꾸기(설명)(py)

입력된 정수의 부호를 바꿔 출력해보자.


예시 
print(-n) 

In [88]:
inputs = int(input())
print(-(inputs))

-1
1


6033 : [기초-산술연산] 문자 1개 입력받아 다음 문자 출력하기(설명)(py)

문자 1개를 입력받아 그 다음 문자를 출력해보자.
영문자 'A'의 다음 문자는 'B'이고, 숫자 '0'의 다음 문자는 '1'이다.

예시
print(chr(n+1))

In [95]:
inputs = input()
print(chr(ord(inputs)+1))

a
b


6034 : [기초-산술연산] 정수 2개 입력받아 차 계산하기(설명)(py)

정수 2개(a, b)를 입력받아 a에서 b를 뺀 차를 출력하는 프로그램을 작성해보자.

예시
c = int(a) - int(b)
print(c)

In [96]:
inputs_01, inputs_02 = input().split(' ')
calculate = int(inputs_01) - int(inputs_02)
print(calculate)

123 -123
246


6035 : [기초-산술연산] 실수 2개 입력받아 곱 계산하기(설명)(py)

실수 2개(f1, f2)를 입력받아 곱을 출력하는 프로그램을 작성해보자.

예시
m = float(f1) * float(f2)
print(m)


In [97]:
inputs_01, inputs_02 = input().split(' ')
calculate = float(inputs_01) * float(inputs_02)
print(calculate)

0.5 2.0
1.0


6036 : [기초-산술연산] 단어 여러 번 출력하기(설명)(py)

단어와 반복 횟수를 입력받아 여러 번 출력해보자.

예시
w, n = input().split()
print(w*int(n))


In [98]:
inputs_01, inputs_02 = input().split(' ')
print(inputs_01 * int(inputs_02))

love 3
lovelovelove


6037 : [기초-산술연산] 문장 여러 번 출력하기(설명)(py)

반복 횟수와 문장을 입력받아 여러 번 출력해보자.

예시
n = input()
s = input()
print(int(n)*s)

In [99]:
inputs_01 = input()
inputs_02 = input()
print(inputs_02 * int(inputs_01))

3
I love CS
I love CSI love CSI love CS


6038 : [기초-산술연산] 정수 2개 입력받아 거듭제곱 계산하기(설명)(py)

정수 2개(a, b)를 입력받아
a를 b번 곱한 거듭제곱을 출력하는 프로그램을 작성해보자.

예시
c = int(a)**int(b) 
print(c)

In [100]:
inputs_01, inputs_02 = input().split(' ')
calculate = int(inputs_01) ** int(inputs_02)
print(calculate)

2 10
1024


6039 : [기초-산술연산] 실수 2개 입력받아 거듭제곱 계산하기(py)

실수 2개(f1, f2)를 입력받아
f1을 f2번 거듭제곱한 값을 출력하는 프로그램을 작성해보자.

In [101]:
inputs_01, inputs_02 = input().split(' ')
calculate = float(inputs_01) ** float(inputs_02)
print(calculate)

4.0 2.0
16.0


6040 : [기초-산술연산] 정수 2개 입력받아 나눈 몫 계산하기(설명)(py)

정수 2개(a, b) 를 입력받아 a를 b로 나눈 몫을 출력해보자.

예시
print(a//b)

In [102]:
inputs_01, inputs_02 = input().split(' ')
calculate = int(inputs_01) // int(inputs_02)
print(calculate)

10 3
3


6041 : [기초-산술연산] 정수 2개 입력받아 나눈 나머지 계산하기(설명)(py)

정수 2개(a, b) 를 입력받아 a를 b로 나눈 나머지를 출력해보자.

예시
print(a%b)

In [103]:
inputs_01, inputs_02 = input().split(' ')
calculate = int(inputs_01) % int(inputs_02)
print(calculate)

10 3
1


6042 : [기초-값변환] 실수 1개 입력받아 소숫점이하 자리 변환하기(설명)(py)

실수 1개를 입력받아
소숫점 이하 두 번째 자리까지의 정확도로 반올림한 값을 출력해보자.


예시

a=float(input())
print( format(a, ".2f") )

In [104]:
inputs = float(input())
print(format(inputs, ".2f"))

3.141592
3.14


6043 : [기초-산술연산] 실수 2개 입력받아 나눈 결과 계산하기(py)

실수 2개(f1, f2)를 입력받아
f1 을 f2 로 나눈 값을 출력해보자. 이 때 소숫점 넷째자리에서 반올림하여 무조건 소숫점 셋째 자리까지 출력한다.

In [107]:
inputs_01, inputs_02 = input().split(' ')
calculate = float(inputs_01) / float(inputs_02)
print(format(calculate, ".3f"))

10.0 3.0
3.333


6044 : [기초-산술연산] 정수 2개 입력받아 자동 계산하기(py)

정수 2개(a, b)를 입력받아 합, 차, 곱, 몫, 나머지, 나눈 값을 자동으로 계산해보자.
단 0 <= a, b <= 2147483647, b는 0이 아니다.

In [110]:
inputs_01, inputs_02 = input().split(' ')

calculate_01 = int(inputs_01) + int(inputs_02)
calculate_02 = int(inputs_01) - int(inputs_02)
calculate_03 = int(inputs_01) * int(inputs_02)
calculate_04 = int(inputs_01) // int(inputs_02)
calculate_05 = int(inputs_01) % int(inputs_02)
calculate_06 = int(inputs_01) / int(inputs_02)

print(calculate_01)
print(calculate_02)
print(calculate_03)
print(calculate_04)
print(calculate_05)
print(format(calculate_06, ".2f"))

10 3
13
7
30
3
1
3.33


6045 : [기초-산술연산] 정수 3개 입력받아 합과 평균 출력하기(설명)(py)

정수 3개를 입력받아 합과 평균을 출력해보자.

In [113]:
inputs_01, inputs_02, inputs_03 = input().split(' ')


calculate_01 = int(inputs_01) + int(inputs_02) +  int(inputs_03)
calculate_02 = calculate_01 / len([inputs_01, inputs_02, inputs_03])

print(calculate_01, format(calculate_02, ".2f"))

1 2 3
6 2.00


6046 : [기초-비트시프트연산] 정수 1개 입력받아 2배 곱해 출력하기(설명)(py)

정수 1개를 입력받아 2배 곱해 출력해보자.

In [114]:
inputs = int(input())
print(inputs<<1)

1024
2048


6047 : [기초-비트시프트연산] 2의 거듭제곱 배로 곱해 출력하기(설명)(py)

정수 2개(a, b)를 입력받아 a를 2b배 곱한 값으로 출력해보자.
0 <= a <= 10, 0 <= b <= 10

예시
a = 2
b = 10
print(a << b)  #210 = 1024 가 출력된다.



In [116]:
inputs_01, inputs_02 = input().split(' ')
print(int(inputs_01) << int(inputs_02))

1 3
8


6048 : [기초-비교연산] 정수 2개 입력받아 비교하기1(설명)(py)

두 정수(a, b)를 입력받아
a가 b보다 작으면 True 를, a가 b보다 크거나 같으면 False 를 출력하는 프로그램을 작성해보자.

예시
print(123<456)  #비교 연산자 < 의 계산 결과인 True(참)가 출력된다.
(123, 456 은 숫자로 작성된 단어이기 때문에 10진수로 인식된다.)

In [117]:
inputs_01, inputs_02 = input().split(' ')
print(int(inputs_01) < int(inputs_02))

1 9
True


6049 : [기초-비교연산] 정수 2개 입력받아 비교하기2(설명)(py)

두 정수(a, b)를 입력받아
a와 b의 값이 같으면 True 를, 같지 않으면 False 를 출력하는 프로그램을 작성해보자.

In [118]:
inputs_01, inputs_02 = input().split(' ')
print(int(inputs_01) == int(inputs_02))

0 0
True


6050 : [기초-비교연산] 정수 2개 입력받아 비교하기3(설명)(py)

두 정수(a, b)를 입력받아
b의 값이 a의 값 보다 크거나 같으면 True 를, 같지 않으면 False 를 출력하는 프로그램을 작성해보자.

In [119]:
inputs_01, inputs_02 = input().split(' ')
print(int(inputs_01) <= int(inputs_02))

0 -1
False


6051 : [기초-비교연산] 정수 2개 입력받아 비교하기4(설명)(py)

두 정수(a, b)를 입력받아
a의 값이 b의 값과 서로 다르면 True 를, 같으면 False 를 출력하는 프로그램을 작성해보자.

In [120]:
inputs_01, inputs_02 = input().split(' ')
print(int(inputs_01) != int(inputs_02))

0 1
True


6052 : [기초-논리연산] 정수 입력받아 참 거짓 평가하기(설명)(py)

정수가 입력되었을 때, True/False 로 평가해주는 프로그램을 작성해보자.

예시
n = int(input())
print(bool(n))

In [130]:
inputs = int(input())
print(bool(int(inputs)))

0
False


6053 : [기초-논리연산] 참 거짓 바꾸기(설명)(py)

정수값이 입력될 때,
그 불 값을 반대로 출력하는 프로그램을 작성해보자.

예시
a = bool(int(input()))
print(not a)


In [131]:
inputs = int(input())
print(not bool(int(inputs)))

1
False


6054 : [기초-논리연산] 둘 다 참일 경우만 참 출력하기(설명)(py)

2개의 정수값이 입력될 때,
그 불 값이 모두 True 일 때에만 True 를 출력하는 프로그램을 작성해보자.

예시
a, b = input().split()
print(bool(int(a)) and bool(int(b)))

In [128]:
inputs_01, inputs_02 = input().split(' ')
print(bool(int(inputs_01)) and bool(int(inputs_02)))

1 0
False


6055 : [기초-논리연산] 하나라도 참이면 참 출력하기(설명)(py)

2개의 정수값이 입력될 때,
그 불 값이 하나라도 True 일 때에만 True 를 출력하는 프로그램을 작성해보자.

In [132]:
inputs_01, inputs_02 = input().split(' ')
print(bool(int(inputs_01)) or bool(int(inputs_02)))

1 1
True


6056 : [기초-논리연산] 참/거짓이 서로 다를 때에만 참

2개의 정수값이 입력될 때,
그 불 값(True/False) 이 서로 다를 때에만 True 를 출력하는 프로그램을 작성해보자.

예시
c = bool(int(a))
d = bool(int(b))
print((c and (not d)) or ((not c) and d))

In [154]:
inputs_01, inputs_02 = input().split(' ')

boolean_01 = bool(int(inputs_01))
boolean_02 = bool(int(inputs_02))

print((boolean_01 and (not boolean_02)) or ((not boolean_01) and boolean_02))

1 1
False


6057 : [기초-논리연산] 참/거짓이 서로 같을 때에만 참 출력하기(설명)(py)

2개의 정수값이 입력될 때,
그 불 값(True/False) 이 서로 같을 때에만 True 를 출력하는 프로그램을 작성해보자.

In [158]:
inputs_01, inputs_02 = input().split(' ')

boolean_01 = bool(int(inputs_01))
boolean_02 = bool(int(inputs_02))

print(((not boolean_01) and (not boolean_02)) or (boolean_01 and boolean_02))

0 1
False


6058 : [기초-논리연산] 둘 다 거짓일 경우만 참 출력하기(py)

2개의 정수값이 입력될 때,
그 불 값(True/False) 이 모두 False 일 때에만 True 를 출력하는 프로그램을 작성해보자.

In [166]:
inputs_01, inputs_02 = input().split(' ')

boolean_01 = bool(int(inputs_01))
boolean_02 = bool(int(inputs_02))

print(not(boolean_01 or boolean_02))

0 0
True


6059 : [기초-비트단위논리연산] 비트단위로 NOT 하여 출력하기(설명)(py)

입력 된 정수를 비트단위로 참/거짓을 바꾼 후 정수로 출력해보자.
비트단위(bitwise)연산자 ~ 를 붙이면 된다.(~ : tilde, 틸드라고 읽는다.)

** 비트단위(bitwise) 연산자는,
~(bitwise not), &(bitwise and), |(bitwise or), ^(bitwise xor),
<<(bitwise left shift), >>(bitwise right shift)
가 있다.

예를 들어 1이 입력되었을 때 저장되는 1을 32비트 2진수로 표현하면
        00000000 00000000 00000000 00000001 이고,
~1은 11111111 11111111 11111111 11111110 가 되는데 이는 -2를 의미한다.

예시
a = 1
print(~a) #-2가 출력된다.

In [173]:
inputs = int(input())
print(~inputs)

2
-3


6060 : [기초-비트단위논리연산] 비트단위로 AND 하여 출력하기(설명)(py)

입력된 정수 두 개를 비트단위로 and 연산한 후 그 결과를 정수로 출력해보자.
비트단위(bitwise)연산자 &를 사용하면 된다.(and, ampersand, 앰퍼센드라고 읽는다.)

In [175]:
inputs_01, inputs_02 = input().split(' ')
print(int(inputs_01) & int(inputs_02))

3 5
1


6061 : [기초-비트단위논리연산] 비트단위로 OR 하여 출력하기(설명)(py)

입력된 정수 두 개를 비트단위로 or 연산한 후 그 결과를 정수로 출력해보자.
비트단위(bitwise) 연산자 |(or, vertical bar, 버티컬바)를 사용하면 된다.

In [176]:
inputs_01, inputs_02 = input().split(' ')
print(int(inputs_01) | int(inputs_02))

3 5
7


6062 : [기초-비트단위논리연산] 비트단위로 XOR 하여 출력하기(설명)(py)

입력된 정수 두 개를 비트단위로 xor 연산한 후 그 결과를 정수로 출력해보자.
비트단위(bitwise) 연산자 ^(xor, circumflex/caret, 서컴플렉스/카릿)를 사용하면 된다.

In [177]:
inputs_01, inputs_02 = input().split(' ')
print(int(inputs_01) ^ int(inputs_02))

3 5
6


6063 : [기초-3항연산] 정수 2개 입력받아 큰 값 출력하기(설명)(py)

입력된 두 정수(a, b) 중 큰 값을 출력하는 프로그램을 작성해보자.
단, 3항 연산을 사용한다.

예시
a, b = input().split()
a = int(a)  #변수 a에 저장되어있는 값을 정수로 바꾸어 다시 변수 a에 저장
b = int(b)
c = (a if (a>=b) else b)
print(int(c))

In [180]:
inputs_01, inputs_02 = input().split(' ')
inputs_01, inputs_02 = int(inputs_01), int(inputs_02)

if_calculate = (inputs_01 if (inputs_01 >= inputs_02) else inputs_02)

print(if_calculate)

123 456
456


6064 : [기초-3항연산] 정수 3개 입력받아 가장 작은 값 출력하기(설명)(py)

입력된 세 정수 a, b, c 중 가장 작은 값을 출력하는 프로그램을 작성해보자.
단, 3항 연산을 사용한다.

In [183]:
a, b, c = input().split(' ')
a, b, c = int(a), int(b), int(c)

if_calculate = (a if a < b else b) if ((a if a < b else b) < c) else c

print(if_calculate)

5 -1 3
-1


6065 : [기초-조건/선택실행구조] 정수 3개 입력받아 짝수만 출력하기(설명)(py)

3개의 정수(a, b, c)가 입력되었을 때, 짝수만 출력해보자.

예시
a, b, c = input().split()

a = int(a)
b = int(b)
c = int(c)

if a%2==0 :  #논리적으로 한 단위로 처리해야하는 경우 콜론(:)을 찍고, 들여쓰기로 작성 한다.
  print(a)

if b%2==0 :
  print(b) 

if c%2==0 :
  print(c) 


In [184]:
inputs_01, inputs_02, inputs_03 = input().split(' ')
inputs_01, inputs_02, inputs_03 = int(inputs_01), int(inputs_02), int(inputs_03)

if inputs_01 % 2 ==0:
    print(inputs_01)

if inputs_02 % 2 ==0:
    print(inputs_02)
    
if inputs_03 % 2 ==0:
    print(inputs_03)

1 2 4
2
4


6066 : [기초-조건/선택실행구조] 정수 3개 입력받아 짝/홀 출력하기(설명)(py)

3개의 정수(a, b, c)가 입력되었을 때, 짝(even)/홀(odd)을 출력해보자.

예시
if a%2==0 :
  print("even")
else :
  print("odd") 

In [187]:
inputs_01, inputs_02, inputs_03 = input().split(' ')
inputs_01, inputs_02, inputs_03 = int(inputs_01), int(inputs_02), int(inputs_03)

if inputs_01 % 2 == 0:
    print("even")
else:
    print("odd")
    
if inputs_02 % 2 == 0:
    print("even")
else:
    print("odd")

if inputs_03 % 2 == 0:
    print("even")
else:
    print("odd")

1 2 1
odd
even
odd


6067 : [기초-조건/선택실행구조] 정수 1개 입력받아 분류하기(설명)(py)

0이 아닌 정수 1개가 입력되었을 때, 음(-)/양(+)과 짝(even)/홀(odd)을 구분해 분류해보자.
음수이면서 짝수이면, A
음수이면서 홀수이면, B
양수이면서 짝수이면, C
양수이면서 홀수이면, D
를 출력한다.

예시
if n<0 :
  if n%2==0 :
    print('A')      #주의 : 변수 A와 문자열 'A' / "A" 는 의미가 완전히 다르다. 
  else :
    print('B')
else :
  if n%2==0 :
    print('C')
  else :
    print('D')

In [189]:
inputs = int(input())

if inputs < 0:
    if inputs % 2 == 0:
        print("A")
    else:
        print("B")
else:
    if inputs % 2 == 0:
        print("C")
    else:
        print("D")

-2147483648
A


6068 : [기초-조건/선택실행구조] 점수 입력받아 평가 출력하기(설명)(py)

점수(정수, 0 ~ 100)를 입력받아 평가를 출력해보자.

평가 기준
점수 범위 : 평가
 90 ~ 100 : A
 70 ~   89 : B
 40 ~   69 : C
   0 ~   39 : D
로 평가되어야 한다.

예시
...
if n>=90 :
  print('A')
else :
  if n>=70 :
    print('B')
  else :
    if n>=40 :
      print('C')
    else :
      print('D') 

In [190]:
inputs = int(input())

if inputs >= 90:
    print("A")
else:
    if inputs >= 70:
        print("B")
    else:
        if inputs >= 40:
            print("C")
        else:
            print("D")

73
B


6069 : [기초-조건/선택실행구조] 평가 입력받아 다르게 출력하기(py)

평가를 문자(A, B, C, D, ...)로 입력받아 내용을 다르게 출력해보자.

평가 내용
평가 : 내용
A : best!!!
B : good!!
C : run!
D : slowly~
나머지 문자들 : what?

In [193]:
inputs = str(input())

if inputs == "A":
    print("best!!!")
else:
    if inputs == "B":
        print("good!!")
    else:
        if inputs == "C":
            print("run!")
        else:
            if inputs == "D":
                print("slowly~")
            else:
                print("what?")

F
what?


6070 : [기초-조건/선택실행구조] 월 입력받아 계절 출력하기(설명)(py)

월이 입력될 때 계절 이름이 출력되도록 해보자.

월 : 계절 이름
12, 1, 2 : winter
  3, 4, 5 : spring
  6, 7, 8 : summer
  9, 10, 11 : fall

예시
if n//3==1 :
  print("spring")

In [195]:
inputs = int(input())

if 3 <= inputs <= 5:
    print("spring")
else:
    if 6 <= inputs <= 8:
        print("summer")
    else:
        if 9 <= inputs <= 11:
            print("fall")
        else:
            print("winter")

12
winter


6071 : [기초-반복실행구조] 0 입력될 때까지 무한 출력하기(설명)(py)

임의의 정수가 줄을 바꿔 계속 입력된다.
-2147483648 ~ +2147483647, 단 개수는 알 수 없다.

0이 아니면 입력된 정수를 출력하고, 0이 입력되면 출력을 중단해보자.

예시
n = 1      #처음 조건 검사를 통과하기 위해 0 아닌 값을 임의로 저장
while n!=0 :
  n = int(input())
  if n!=0 :
    print(n)

In [197]:
inputs = 1

while inputs != 0:
    inputs = int(input())
    
    if inputs != 0:
        print(inputs)
    else:
        break

7
7
4
4
2
2
3
3
0


6072 : [기초-반복실행구조] 정수 1개 입력받아 카운트다운 출력하기1(설명)(py)

정수(1 ~ 100) 1개가 입력되었을 때 카운트다운을 출력해보자.

while 조건식 :
반복 실행구조를 사용해 보자.

예시
while n!=0 :
  print(n)
  n = n-1

In [198]:
inputs = int(input())

while inputs != 0:
    print(inputs)
    inputs -= 1

5
5
4
3
2
1


6073 : [기초-반복실행구조] 정수 1개 입력받아 카운트다운 출력하기2(py)

정수(1 ~ 100) 1개가 입력되었을 때 카운트다운을 출력해보자.

while 조건식 :
  ...
  ...

반복 실행구조를 사용해 보자.

In [202]:
inputs = int(input())

while inputs != 0:
    if inputs >= 0:
        inputs -= 1
        print(inputs)
    else:
        break

5
4
3
2
1
0


6074 : [기초-반복실행구조] 문자 1개 입력받아 알파벳 출력하기(설명)(py)

영문 소문자(a ~ z) 1개가 입력되었을 때,
a부터 그 문자까지의 알파벳을 순서대로 출력해보자.

예시
c = ord(input())
t = ord('a')
while t<=c :
  print(chr(t), end=' ')
  t += 1

In [207]:
inputs = input()

start_number = ord('a')
end_number = ord(inputs)

while start_number <= end_number :
    print(chr(start_number), end = ' ')
    start_number += 1 

f
a b c d e f 

6075 : [기초-반복실행구조] 정수 1개 입력받아 그 수까지 출력하기1(py)

정수(0 ~ 100) 1개를 입력받아 0부터 그 수까지 순서대로 출력해보자.

In [208]:
inputs = input()

start_number = 0
end_number = int(inputs)

while start_number <= end_number :
    print(start_number)
    start_number += 1 

5
0
1
2
3
4
5


6076 : [기초-반복실행구조] 정수 1개 입력받아 그 수까지 출력하기2(설명)(py)

정수(0 ~ 100) 1개를 입력받아 0부터 그 수까지 순서대로 출력해보자.

예시
n = int(input())
for i in range(n+1) :
  print(i)

In [209]:
inputs = int(input())

for i in range(inputs+1):
    print(i)

5
0
1
2
3
4
5


6077 : [기초-종합] 짝수 합 구하기(설명)(py)

정수(1 ~ 100) 1개를 입력받아 1부터 그 수까지 짝수의 합을 구해보자.

예시
#다음 코드는 홀 수만 더해 출력한다.
n = int(input())
s = 0
for i in range(1, n+1) :
  if i%2==1 :
    s += i

print(s)

In [214]:
inputs = int(input())

Sum_Number = 0

for i in range(1, inputs+1):
    if i % 2 == 0:
        Sum_Number += i
        
print(Sum_Number)

23
132


6078 : [기초-종합] 원하는 문자가 입력될 때까지 반복 출력하기(py)

영문 소문자 'q'가 입력될 때까지
입력한 문자를 계속 출력하는 프로그램을 작성해보자.

In [222]:
while True:
    inputs = str(input())
    if inputs != 'q':
        print(inputs)
    else:
        print(inputs)
        break

x
x
b
b
z
z
q
q


6079 : [기초-종합] 언제까지 더해야 할까?(py)

1, 2, 3 ... 을 계속 더해 나갈 때,
그 합이 입력한 정수(0 ~ 1000)보다 같거나 작을 때까지만
계속 더하는 프로그램을 작성해보자.

즉, 1부터 n까지 정수를 계속 더해 나간다고 할 때,
어디까지 더해야 입력한 수보다 같거나 커지는 지를 알아보고자하는 문제이다.

In [313]:
inputs = int(input())

Result = 0
i = 1

while True:
    if Result < inputs:
        Result += i
        
        
    else:
        Result -= i - 1
        i = i - 1
        break
    i = i + 1
print(i)

55
10


6080 : [기초-종합] 주사위 2개 던지기(설명)(py)

1부터 n까지, 1부터 m까지 숫자가 적힌 서로 다른 주사위 2개를 던졌을 때,
나올 수 있는 모든 경우를 출력해보자.

예시
for i in range(1, n+1) :
  for j in range(1, m+1) :
    print(i, j)

In [319]:
inputs_01, inputs_02 = input().split(' ')

for i in range(1, int(inputs_01) + 1):
    for j in range(1, int(inputs_02) + 1):
        print(i, j)

2 3
1 1
1 2
1 3
2 1
2 2
2 3


6081 : [기초-종합] 16진수 구구단 출력하기(py)

16진수(0, 1, 2, 3, 4, 5, 6, 7, 8, 9, A, B, C, D, E, F)를 배운
영일이는 16진수끼리 곱하는 16진수 구구단?에 대해서 궁금해졌다.

A, B, C, D, E, F 중 하나가 입력될 때,
1부터 F까지 곱한 16진수 구구단의 내용을 출력해보자.
(단, A ~ F 까지만 입력된다.)

예시
  print('%X'%n, '*%X'%i, '=%X'%(n*i), sep='')

In [63]:
inputs = int(input(), 16)
for i in range(1, 16):
    print('%X'%inputs,"*", '%X'%i, '=%X'%(inputs*i), sep='')

B
B*1=B
B*2=16
B*3=21
B*4=2C
B*5=37
B*6=42
B*7=4D
B*8=58
B*9=63
B*A=6E
B*B=79
B*C=84
B*D=8F
B*E=9A
B*F=A5


6082 : [기초-종합] 3 6 9 게임의 왕이 되자(설명)(py)

친구들과 함께 3 6 9 게임을 하던 영일이는 잦은 실수 때문에 계속해서 벌칙을 받게 되었다.
3 6 9 게임의 왕이 되기 위한 369 마스터 프로그램을 작성해 보자.

** 3 6 9 게임은?
여러 사람이 순서를 정한 후, 순서대로 수를 부르는 게임이다.
만약 3, 6, 9 가 들어간 수를 자신이 불러야 하는 상황이라면, 수를 부르는 대신 "박수(X)" 를 쳐야 한다.
33과 같이 3,6,9가 두 번 들어간 수 일때, "짝짝"과 같이 박수를 두 번 치는 형태도 있다. 


입력이 1~ 30까지의 정수일 때

In [ ]:
    if i % 3 == 0:
        print("X", end=' ')
    else:

In [2]:
inputs = int(input())

Result = []
for i in range(1, inputs+1):
    Result.append(i)
    
print(Result)

9
[1, 2, 3, 4, 5, 6, 7, 8, 9]


In [13]:
inputs = int(input())

for i in range(1, inputs+1):
    if i % 10 == 3:
        print("X", end = ' ')
    elif i % 10 == 6 :
        print("X", end = ' ')
    elif i % 10 == 9 :
        print("X", end = ' ')
    else:
        print(i, end = ' ')

36
1 2 X 4 5 X 7 8 X 10 11 12 X 14 15 X 17 18 X 20 21 22 X 24 25 X 27 28 X 30 31 32 X 34 35 X 

6083 : [기초-종합] 빛 섞어 색 만들기(설명)(py)

빨강(red), 초록(green), 파랑(blue) 빛을 섞어 여러 가지 다른 색 빛을 만들어 내려고 한다.

빨강(r), 초록(g), 파랑(b) 각 빛의 가짓수가 주어질 때,
주어진 rgb 빛들을 섞어 만들 수 있는 모든 경우의 조합(r g b)과 만들 수 있는 색의 가짓 수를 계산해보자.  

In [33]:
r, g, b = map(int, input().split(' '))

for i in range(r):
    for j in range(g):
        for k in range(b):
            print(i, j, k)

print(r*g*b)

2 2 2
0 0 0
0 0 1
0 1 0
0 1 1
1 0 0
1 0 1
1 1 0
1 1 1
8


6084 : [기초-종합] 소리 파일 저장용량 계산하기(py)

녹음할 시간(초) s가 주어질 때,

필요한 저장 용량을 계산하는 프로그램을 작성해보자.


In [54]:
h, b, c, s = map(int, input().split(' '))

print(round(((((h * b * c * s) / 8 ) / 1024) / 1024), 1), end=" MB")

44100 16 2 10
1.7 MB

6085 : [기초-종합] 그림 파일 저장용량 계산하기(py)

이미지의 가로 해상도 w, 세로 해상도 h, 한 픽셀을 저장하기 위한 비트 b 가 주어질 때,
압축하지 않고 저장하기 위해 필요한 저장 용량을 계산하는 프로그램을 작성해 보자.

In [63]:
w, h, b = map(int, input().split(' '))

Calculate = ((((w * h * b) / 8 ) / 1024) / 1024)
print("{0:.2f}".format(Calculate), end=" MB")

100 100 4
0.00 MB

6086 : [기초-종합] 거기까지! 이제 그만~(설명)(py)

1, 2, 3 ... 을 순서대로 계속 더해 합을 만드는데,
그 합이 입력한 정수보다 작을 동안만 계속 더하는 프로그램을 작성해보자.

In [69]:
inputs = int(input())

Start = 0
Result = 0

while True:
    Start += 1
    Result += Start
    if Result >= inputs:
        break
        
print(Result)

33
36


6087 : [기초-종합] 3의 배수는 통과(설명)(py)

1부터 입력한 정수까지 1씩 증가시켜 출력하는 프로그램을 작성하되,
3의 배수인 경우는 출력하지 않도록 만들어보자.

In [72]:
inputs = int(input())

for i in range(1, inputs+1):
    if i % 3 == 0:
        continue
    print(i, end = " ")

10
1 2 4 5 7 8 10 

6088 : [기초-종합] 수 나열하기1(py)

시작 값(a), 등차(d), 몇 번째인지를 나타내는 정수(n)가 입력될 때
n번째 수를 출력하는 프로그램을 만들어보자.


풀이 1

In [78]:
a, d, n = map(int, input().split(' '))

print(d * (n-1) + a)

1 3 5
13


풀이 2

In [2]:
a, d, n = map(int, input().split(' '))

for i in range(1, n):
    a += d
    
print(a)

1 3 2
4


6089 : [기초-종합] 수 나열하기2(py)

시작 값(a), 등비(r), 몇 번째인지를 나타내는 정수(n)가 입력될 때
n번째 수를 출력하는 프로그램을 만들어보자.

In [93]:
a, r, n = map(int, input().split(' '))

for i in range(1, n):
    a *= r
    
print(a)

2 3 7
1458


6090 : [기초-종합] 수 나열하기3(py)

시작 값(a), 곱할 값(m), 더할 값(d), 몇 번째인지를 나타내는 정수(n)가 입력될 때,
n번째 수를 출력하는 프로그램을 만들어보자.

In [94]:
a, m, d, n = map(int, input().split(' '))

for i in range(1, n):
    a *= m
    a += d
    
print(a)

1 -2 1 8
-85


6091 : [기초-종합] 함께 문제 푸는 날(설명)(py)

같은 날 동시에 가입한 3명의 사람들이 온라인 채점시스템에 들어와 문제를 푸는 날짜가
매우 규칙적이라고 할 때, 다시 모두 함께 문제를 풀게 되는 그날은 언제일까?

In [96]:
a, b, c = map(int, input().split(' '))

Day = 1

while Day % a != 0 or Day % b != 0 or Day % c !=0:
    Day +=1
print(Day)

3 7 9
63


6092 : [기초-리스트] 이상한 출석 번호 부르기1(설명)(py)

출석 번호를 n번 무작위로 불렀을 때, 각 번호(1 ~ 23)가 불린 횟수를 각각 출력해보자.

In [138]:
inputs_01 = int(input())
inputs_02 = list(map(int, input().split(' ')))

Temp = []
for i in range(24):
    Temp.append(0)
    
for j in range(inputs_01):
    Temp[inputs_02[j]] += 1
    
for k in range(1, 24):
    print(Temp[k], end = " ")

1000
9 20 11 3 11 2 21 7 22 22 6 6 1 9 16 10 15 12 21 11 8 21 16 8 15 18 4 5 10 20 18 20 20 11 13 2 9 1 3 11 15 18 4 22 23 15 21 22 9 14 16 21 16 14 13 17 8 21 19 14 14 8 19 12 8 9 15 19 14 8 13 15 18 6 14 22 15 8 22 7 1 6 21 21 20 21 16 13 19 10 7 8 17 12 18 15 10 7 14 11 4 2 21 23 13 23 16 22 17 10 22 18 21 13 7 7 4 13 2 9 7 13 8 23 21 14 14 1 2 12 9 15 1 10 9 16 1 9 17 2 8 17 6 14 14 9 17 4 23 7 4 2 5 8 6 23 14 1 17 10 14 7 10 1 23 17 8 14 11 12 8 6 22 22 17 21 20 23 23 18 18 12 1 15 16 17 11 23 14 20 11 19 16 2 17 19 16 15 11 4 5 22 4 12 1 21 18 17 21 10 7 4 19 2 1 12 22 11 9 23 18 18 14 23 15 7 21 13 21 23 3 13 22 2 4 13 18 11 16 19 5 20 4 18 22 7 17 12 13 16 9 6 7 16 20 15 20 1 13 22 19 17 9 1 17 6 17 15 1 22 3 10 16 11 5 14 23 12 21 9 23 21 22 14 8 20 16 3 11 21 9 2 18 13 7 6 15 17 20 9 9 12 11 19 14 3 21 5 8 21 9 4 22 13 3 20 4 18 13 11 3 4 16 14 9 21 6 2 17 3 1 6 9 6 10 3 20 18 15 10 13 19 8 8 5 8 19 15 9 12 23 8 16 8 11 10 16 2 18 23 4 23 19 2 15 5 3 11 10 18 13 4 10 12 15 14

모범답안

In [ ]:
n = int(input())
a = input().split()

for i in range(n):
    a[i] = int(a[i])

d = []                     
for i in range(24):
    d.append(0)

for i in range(n):
    d[a[i]] += 1

for i in range(1, 24):
    print(d[i], end=' ')

6093 : [기초-리스트] 이상한 출석 번호 부르기2(py)

출석 번호를 n번 무작위로 불렀을 때, 부른 번호를 거꾸로 출력해 보자.

In [144]:
n = int(input())
a = input().split()

for i in range(n-1, -1, -1) :
    print(a[i], end=' ')

10
10 4 2 3 6 6 7 9 8 5
5 8 9 7 6 6 3 2 4 10 

6094 : [기초-리스트] 이상한 출석 번호 부르기3(py)

출석 번호를 n번 무작위로 불렀을 때, 가장 빠른 번호를 출력해 보자.

In [152]:
n = int(input())
a = input().split()

for i in range(n):
    a[i] = int(a[i])

print(min(list(map(int, a))))

10
10 4 2 3 6 6 7 9 8 5
[10, 4, 2, 3, 6, 6, 7, 9, 8, 5]
2


6095 : [기초-리스트] 바둑판에 흰 돌 놓기(설명)(py)

바둑판(19 * 19)에 n개의 흰 돌을 놓는다고 할 때,
n개의 흰 돌이 놓인 위치를 출력하는 프로그램을 작성해보자.

In [37]:
N = int(input())

Field = []
for i in range(20):
    Field.append([])
    for j in range(20): 
        Field[i].append(0)


for k in range(N):
    X, Y = map(int, input().split(' '))
    Field[X-1][Y-1] = 1
        

for i in range(19):
    for j in range(19):
        print(Field[i][j], end = ' ')
    print()

5
1 1
2 2
3 3
4 4
5 5
1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 
0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 
0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 
0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 
0 0 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 
0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 


6096 : [기초-리스트] 바둑알 십자 뒤집기(py)

바둑판(19 * 19)에 흰 돌(1) 또는 검정 돌(0)이 모두 꽉 채워져 놓여있을 때,
n개의 좌표를 입력받아 십(+)자 뒤집기한 결과를 출력하는 프로그램을 작성해보자.

In [ ]:
Field = []
for k in range(19):
    inputs = list(map(int, input().split(' ')))
    Field.append(inputs)

N = int(input())
for l in range(N):
    X, Y = map(int, input().split(' '))
    for i in range(19):
        if Field[i][Y-1]==0:
            Field[i][Y-1]=1
        else:
            Field[i][Y-1]=0
        
        if Field[X-1][i]==0:
            Field[X-1][i]=1
        else:
            Field[X-1][i]=0
            
for i in range(19):
    for j in range(19):
        print(Field[i][j], end = ' ')
    print()

6097 : [기초-리스트] 설탕과자 뽑기(py)

격자판을 채운 막대의 모양을 출력하는 프로그램을 만들어보자.

In [94]:
H, W = map(int, input().split(' '))

Field = []
for i in range(H):
    Field.append([])
    for j in range(W): 
        Field[i].append(0)
        
N = int(input())
for k in range(N):
    L, D, X, Y = map(int, input().split(' '))
    
    if D==0:    
        for l in range(L):
            Field[X-1][Y-1+l] = 1
    elif D==1:
        for l in range(L):
            Field[X-1+l][Y-1] = 1
            
for m in range(H):
    for n in range(W):
        print(Field[m][n], end = ' ')
    print()

5 5
3
2 0 1 1
3 1 2 3
4 1 2 5


6098 : [기초-리스트] 성실한 개미(py)

 성실한 개미의 이동 경로를 예상해보자.

In [ ]:
Field = []
for k in range(10):
    inputs = list(map(int, input().split(' ')))
    Field.append(inputs)

i, j = 1, 1

while True:

    if Field[i][j] == 0:
        Field[i][j] = 9
    
    elif Field[i][j] == 2:
        Field[i][j] = 9
        break
    
    if Field[i][j+1] == 1 and Field[i+1][j] == 1:
        Field[i][j] = 9
        break
        
    if Field[i][j+1] == 1:
        i += 1
    elif Field[i+1][j] == 1:
        j += 1
    else:
        j += 1
    
    
for m in range(10):
    for n in range(10):
        print(Field[m][n], end = ' ')
    print()